contracts/staking/Staking.sol

```diff
@@ -1563,9 +1563,43 @@ contract Staking is StakingV1Storage, GraphUpgradeable, IStaking {
         uint256 delegationRewards = _collectDelegationIndexingRewards(_indexer, totalRewards);
         uint256 indexerRewards = totalRewards.sub(delegationRewards);
 
-        // Add the rest of the rewards to the indexer stake
-        if (indexerRewards > 0) {
-            _stake(_indexer, indexerRewards);
+        // Send the indexer rewards
+        _sendRewards(
+            graphToken(),
+            indexerRewards,
+            _indexer,
+            rewardsDestination[_indexer] == address(0)
+        );
+    }
+
+    /**
+     * @dev Send rewards to the appropiate destination.
+     * @param _graphToken Graph token
+     * @param _amount Number of rewards tokens
+     * @param _beneficiary Address of the beneficiary of rewards
+     * @param _restake Whether to restake or not
+     */
+    function _sendRewards(
+        IGraphToken _graphToken,
+        uint256 _amount,
+        address _beneficiary,
+        bool _restake
+    ) private {
+        if (_amount == 0) return;
+
+        if (_restake) {
+            // Restake to place fees into the indexer stake
+            _stake(_beneficiary, _amount);
+        } else {
+            // Transfer funds to the beneficiary's designated rewards destination if set
+            address destination = rewardsDestination[_beneficiary];
+            require(
+                _graphToken.transfer(
+                    destination == address(0) ? _beneficiary : destination,
```

Comment:
I believe this ternary is not needed. If we got this far, we already know that `rewardsDestination[_beneficiary] != 0`

this would work:
```
require(_graphToken.transfer(rewardsDestination[beneficiary]), _amount, "!transfer");
```
wdyt?

Reply:
It depends, when the function is called from _claim() it might not be the case.

Reply:
you are right ! there is nothing wrong with this code